src/plugins/auto/auto_request_wrapper.hpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include <atomic>
+#include <mutex>
+#include <unordered_map>
+#include <map>
+#include <vector>
+#include <string>
+
+#include "cpp_interfaces/impl/ie_executable_network_thread_safe_default.hpp"
+#include "threading/ie_thread_safe_containers.hpp"
+#include "threading/ie_itask_executor.hpp"
+#include "threading/ie_executor_manager.hpp"
+#include "ie_icore.hpp"
+#include "request_wrapper.hpp"
+#ifdef  MULTIUNITTEST
+#define MOCKTESTMACRO virtual
+#define MultiDevicePlugin MockMultiDevicePlugin
+#else
+#define MOCKTESTMACRO
+#endif
+
+namespace MultiDevicePlugin {
+
+class MultiDeviceInferencePlugin;
+class AutoRequestWrapper : public RequestWrapper  {
```

Comment:
@myshevts  abstracting the createRequest is more for below consideration :
let's say if we are attempting to enable some new kind of schedule logic (for example, if we want to bind the auto infer request to hw request and then avoid the memcpy and also the potential setBlobs in auto ), then it would be easier for us to implement the new request wrapper ,e.g a new async infer request with new pipeline, without interrupting the existing scheduler logic (which can continue to be used in multi, or always as a baseline for future changes)

Reply:
@myshevts , how about this simplified version https://github.com/openvinotoolkit/openvino/pull/11253
we can make the change under the current code structure first and improve further.